src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -2544,9 +2544,17 @@ void MacroAssembler::debug64(char* msg, int64_t pc, int64_t regs[])
   }
 }
 
+RegSet MacroAssembler::call_clobbered_registers() {
+  RegSet regs = RegSet::range(r0, r17) - RegSet::of(rscratch1, rscratch2);
+#ifndef R18_RESERVED
+  regs += r18_reserved;
+#endif
```

Comment:
Is R18 really call clobbered? And it is Java's job to save and restore it? I would not have though it was, but I don't know the calling convention. Either way, someone should carefully review this.